<launch>
    <arg name="robot_type" default="ylo2" doc="Robot type: [ylo2, a1, aliengo, go1, laikago]"/>

    <!-- Launch the imu node -->
    <node pkg="ylo2_imu" type="ylo2_imu" name="ylo2_imu">
        <param name="port" value="/dev/ttyACM0" />
        <param name="baud_rate" value="115200" />
        <param name="frame_id" value="ylo2_imu" />
        <param name="use_gui" value="true" />
    </node>
    
    <!-- Call Ylo2 robot under A1 naming -->
    <param name="legged_robot_description" command="$(find xacro)/xacro $(find legged_ylo2_description)/urdf/ylo2.xacro
       robot_type:=$(arg robot_type) "/>

    
    
    <node name="generate_urdf" pkg="legged_common" type="generate_urdf.sh" output="screen"
          args="$(find legged_ylo2_description)/urdf/ylo2.xacro $(arg robot_type)"/>

    <rosparam file="$(find legged_ylo2_hw)/config/$(arg robot_type).yaml" command="load"/>

    <param name="robot_type" value="$(arg robot_type)"/>
    <node name="legged_ylo2_hw" pkg="legged_ylo2_hw" type="legged_ylo2_hw" respawn="false"
          clear_params="true" output="screen" />

    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find legged_ylo2_description)/launch/hardware.rviz" required="true"/>

</launch>
